### PR TITLE
initial commit for 2.0 support

### PR DESCRIPTION
**Pull request by nkolba**
- added 2.0 API side by side with 1.2
- stubbed out 2.0 in preload
- internalized type dependencies for 1.2

### CODE REVIEW

**Comment by nkolba** [comment]
WIP.  Ran into problems with type usage conflicts between 1.2 and 2.0.

**Comment by robmoffat** [comment]
> WIP. Ran into problems with type usage conflicts between 1.2 and 2.0.

Looks really promising nonetheless!  Looking forward to chatting about it later

**Comment by sebbenmbarek** [comment]
What he said ☝️

**Review by sebbenmbarek — APPROVED** [review]
LGTM
I like the path aliasing, it makes it so much cleaner

I take it that FDC3Events will be split between 1.2 and 2 in a later iteration?

File: package.json [hunk 1]
```diff
@@ -61,12 +62,14 @@
...
     "@emotion/styled": "^11.8.1",
     "@finos/fdc3": "2.0.0-beta.2",
+    "@finos/fdc3-1.2": "@finos/fdc3@1.2.0",
     "@fontsource/roboto": "^4.5.3",
     "@mui/icons-material": "^5.4.4",
     "@mui/material": "^5.4.4",
     "electron-fetch": "^1.7.4",
     "electron-updater": "^4.6.5",
     "fastify": "^3.27.2",
+    "fdc3-1.2": "npm:@finos/fdc3@^1.2.0",
```
Comment (robmoffat) [inline package.json:72]: should this be here again?

File: packages/main/src/IntentResolver.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import { View } from './view';
 import { getRuntime } from './index';
 import { BrowserWindow } from 'electron';
-import { FDC3App, IntentInstance, ResolverDetail } from './types/FDC3Data';
+import {
```
Comment (robmoffat) [inline packages/main/src/IntentResolver.ts:4]: Is IntentResolver common to 1.2 and 2.0?

File: packages/preload/tsconfig.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
 
     "baseUrl": ".",
     "paths": {
-      "/@/*": ["./src/*"]
+      "/@/*": ["./src/*"],
+      "/@main/*": ["../main/src/*"],
```
Comment (robmoffat) [inline packages/preload/tsconfig.json:18]: I think it's usual to leave off the / on `/@main`, no?

**Review by robmoffat — APPROVED** [review]

**Pull request merged**